src/chat.rs

```diff
@@ -650,11 +661,16 @@ impl Chat {
             Some(message) => message.text.unwrap_or_else(String::new),
             _ => String::new(),
         };
+        let archive_state = self.id.get_archive_state(context);
         Ok(ChatInfo {
             id: self.id,
             type_: self.typ as u32,
             name: self.name.clone(),
-            archived: self.archived,
+            archived: match archive_state {
+                ArchiveState::Normal => 0,
+                ArchiveState::Archived => 1,
+                ArchiveState::Pinned => 2,
```

Comment:
why is `archived` not an `ArchiveState`?

Reply:
Because @r10s said the naming should be consistent with the other ffi-functions. Even though we agreed that archived is not the best name for this property.

Reply:
my comment was about the naming in the ffi, not about the type in rust :)